spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/session/SessionAutoConfigurationJdbcTests.java

```diff
@@ -86,4 +88,15 @@ public void customTableName() {
 				.queryForList("select * from FOO_BAR")).isEmpty();
 	}
 
+	@Test
+	public void customTableNameWithDefaultSchemaFails() {
+		this.thrown.expect(BeanCreationException.class);
+		this.thrown.expectMessage(containsString(
+				"Database initializer cannot be configured"));
+		load(Arrays.asList(EmbeddedDataSourceConfiguration.class,
+				DataSourceTransactionManagerAutoConfiguration.class),
+				"spring.session.store-type=jdbc",
+				"spring.session.jdbc.table-name=FOO_BAR");
```

Comment:
Thanks for the feedback.

> So basically you're saying you want JDBC and a custom table name and you'll get an exception?

No, it's an exception when you configure custom table name **together** with default schema script location, those two together form an invalid configuration.

> I would rather move the initialization of the table conditional on whether a custom table name has been specified or not.

Solely on whether a custom table name has been specified or not? Configuring custom table name together with appropriate schema script is a valid configuration with existing test in `SessionAutoConfigurationJdbcTests#customTableName` to support it. Spring Batch also has the similar test.

> You provided a custom table name? Then you're on your own to create the schema (instead of throwing an exception as this PR does).

Well, not unless you also customize the schema script location, as explained above.

> You provided the path to a script? We'll of course use that to initialize the database

:+1: 

> You didn't provide anything? We're going to assume you want us to create the schema with the default table name.

:+1:

To address your [first comment](https://github.com/spring-projects/spring-boot/pull/6649#discussion_r74966771) here, I've initially considered placing the logic in the `SessionProperties` itself, but wasn't sure if that was the right place to do it.

I've also considered simply disabling the initializer when configuring custom table name and leaving the default schema script location, but ultimately I've felt that it would be like a silent failure that can easily blow up in runtime so it looked better to me to fail fast.

I'm all for reworking the PR but I don't think the support for custom table name combined with custom schema should be dropped.